src/components/elements/Editor.vue

```diff
@@ -22,35 +22,26 @@ const props = defineProps({
 
 const emit = defineEmits(['update:modelValue'])
 
-const editor = ref<any>(null)
-
-watch(props, newProps => {
-    const isSame = editor.value?.getHTML() === newProps.modelValue
-    if (isSame) {
-      return
-    }
-    editor.value?.commands.setContent(newProps.modelValue, false)
-  }
-)
-
-onMounted(() => {
-  editor.value = new Editor({
-    extensions: [
-      Document,
-      Paragraph,
-      Text,
-      Bold,
-      Italic,
-      History,
-    ],
-    content: props.modelValue,
-    onUpdate: () => {
-      emit('update:modelValue', editor.value?.getHTML().replaceAll(/\<br.*?\>/g, ''))
-    },
-  })
+const editor = useEditor({
+  extensions: [
+    Document,
+    Paragraph,
+    Text,
+    Bold,
+    Italic,
+    History,
+  ],
+  content: props.modelValue,
+  onUpdate: () => {
+    emit('update:modelValue', editor.value?.getHTML().replaceAll(/\<br.*?\>/g, ''))
+  },
 })
 
-onBeforeUnmount(() => {
-  editor.value?.destroy()
```

Comment:
Was this intentionally deleted? I'm less familiar with tiptap - will there be issues with not explicitly destroying the editor if the block is deleted or if the block type is changed?

Reply:
I think it will be handled internally by the composable function `useEditor`.

https://github.com/ueberdosis/tiptap/blob/e2f8747e2b10f98d49798b15d64db5f0a36a3427/packages/vue-3/src/useEditor.ts#L13-L15